internal/app/starter/master_linux_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package starter
+
+import (
+	"testing"
+
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+func TestDoRPCThread(t *testing.T) {
+	test.DropPrivilege(t)
+	defer test.ResetPrivilege(t)
+
+	var fatal error
+	fatalChan := make(chan error, 1)
+
+	var fakeEngine engines.Engine
+
+	tests := []struct {
+		name         string
+		rpcSocket    int
+		containerPid int
+		engine       *engines.Engine
+		shallPass    bool
+	}{
+		{
+			name:         "invalid case",
+			rpcSocket:    -1,
+			containerPid: -1,
+			engine:       nil,
+			shallPass:    false,
+		},
+		{
+			name:         "fake engine struct",
+			rpcSocket:    -1,
+			containerPid: -1,
+			engine:       &fakeEngine,
```

Comment:
please create a new engine (`&engines.Engine{}` or `new(engines.Engine)`) for each test case, otherwise the code under test might end up modifying the engine for the next test.